frontend/setupChelonia.js

```diff
@@ -297,4 +299,15 @@ const setupChelonia = async (): Promise<*> => {
   })
 }
 
-export default setupChelonia
+export default ((() => {
+  let promise
+  return () => {
+    if (!promise) {
+      promise = setupChelonia().catch((e) => {
+        promise = undefined
+        throw e
+      })
+    }
+    return promise
+  }
+})(): () => Promise<void>)
```

Comment:
🚨🥴 Weirdness!

```
if (needed(weirdness)) {
  pleaseAdd(comment)
}
```

Reply:
Noted. This is a recent addition I was trying out. It's meant to make the Promise a singleton, though I can document it and also try to make it more readable if possible.